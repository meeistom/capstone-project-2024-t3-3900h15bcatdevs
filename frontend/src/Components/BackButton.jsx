import React from "react";
import { Button } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowLeft } from "@fortawesome/free-solid-svg-icons";

import "bootstrap/dist/css/bootstrap.min.css";
import "../index.css";

export { BackButton };

function BackButton({ id, onClick, children }) {
  return (
    <Button id={id} variant="outline-secondary" size="lg" onClick={onClick}>
      <FontAwesomeIcon icon={faArrowLeft} />
      <span className="ms-2">{children}</span>
    </Button>
  );
}
