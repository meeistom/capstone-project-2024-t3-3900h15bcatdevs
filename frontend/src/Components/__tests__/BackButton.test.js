import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BackButton } from "../BackButton";
import "@testing-library/jest-dom";

afterEach(() => {
  cleanup();
});

test("should render back button", () => {
  render(<BackButton onClick={null}>Hello</BackButton>);
  const backButton = screen.getByRole("button", { name: "back-button" });

  expect(backButton).toBeInTheDocument();
});

test("should match snapshot", () => {
  const tree = renderer
    .create(<BackButton onClick={null}>Hello</BackButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("should pass down children", () => {
  render(<BackButton onClick={null}>Hello</BackButton>);
  const backButton = screen.getByRole("button", { name: "back-button" });

  expect(backButton).toHaveTextContent("Hello");
});

test("should handle onClick", () => {
  const onClick = jest.fn();

  render(<BackButton onClick={onClick}>Hello</BackButton>);
  const backButton = screen.getByRole("button", { name: "back-button" });

  fireEvent.click(backButton);
  expect(onClick).toHaveBeenCalledTimes(1);
});
