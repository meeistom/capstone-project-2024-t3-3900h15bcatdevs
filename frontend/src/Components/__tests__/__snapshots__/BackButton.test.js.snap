// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  aria-label="back-button"
  className="btn btn-outline-secondary btn-lg"
  disabled={false}
  onClick={null}
  type="button"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-left "
    data-icon="arrow-left"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="ms-2"
  >
    Hello
  </span>
</button>
`;
